src/v/coproc/wasm_event.cc

```diff
@@ -59,6 +71,24 @@ get_value_for_event_header(const model::record& r, event_header header) {
       [&match](const model::record_header& rh) { return rh.key() == match; });
 }
 
+std::variant<wasm::errc, coproc_type> get_coproc_type(const model::record& r) {
+    auto itr = get_value_for_event_header(r, event_header::type);
+    if (itr == r.headers().cend()) {
+        return coproc_type::nodejs; // support old msgs from topic
+    }
+
+    const auto& coproc_type_raw = itr->value();
+    auto coproc_type = iobuf_const_parser(coproc_type_raw)
+                         .read_bytes(coproc_type_raw.size_bytes());
+    if (coproc_type.empty()) {
+        return coproc_type::nodejs;
```

Comment:
I think this should be an error. If the header doesn't exist (supporting old messages) we can turn to a default, but if the header exists and theres just no value, we should decide that this is an error in rpk and not choose a default. So we can return `expected_coproc_type` here

Reply:
done